angel_system/global_step_prediction/global_step_predictor.py

```diff
@@ -464,6 +472,48 @@ def process_new_confidences(self, activity_confs):
 
         return self.trackers
 
+    def add_skipped_granular_step(self, tracker_ind, granular_step):
+        self.trackers[tracker_ind]["skipped_granular_steps"].append(granular_step)
+
+    def get_skipped_steps_one_tracker(self, tracker_ind):
+        tracker = self.trackers[tracker_ind]
+        skipped_steps = []
+        for granular_step in tracker["skipped_granular_steps"]:
+            skipped_steps.append(
+                {
+                    "recipe": tracker["recipe"],
+                    "granular": granular_step,
+                    "part_of_broad": self.granular_to_broad_step(
+                        tracker, granular_step
+                    ),
+                    "activity_id": self.get_activity_from_granular_step(
+                        tracker, granular_step
+                    )[0],
+                    "activity_str": self.get_activity_from_granular_step(
+                        tracker, granular_step
+                    )[1],
+                }
+            )
+        return skipped_steps
+
+    def get_activity_from_granular_step(self, tracker, granular_step):
+        activity_id = tracker["granular_step_to_activity_id"][granular_step]
+        activity_str = self.get_activity_str_from_id(activity_id)
+
+        return activity_id, activity_str
+
+    def get_activity_str_from_id(self, activity_id):
+        return self.activity_config["labels"][activity_id]["full_str"]
+        fdsao
```

Comment:
should this be here?

Reply:
Interersting. No it shouldn't. Thanks for the check - removing.

Reply:
Removed